library/core/src/iter/adapters/enumerate.rs

```diff
@@ -15,10 +16,131 @@ use crate::ops::{Add, AddAssign, Try};
 pub struct Enumerate<I> {
     iter: I,
     count: usize,
+    len: usize,
 }
-impl<I> Enumerate<I> {
+impl<I: Iterator> Enumerate<I> {
```

Comment:
Adding a trait bound here is technically a breaking change. Before this change, `Enumerate<T>` can be 'used' in code that's generic over `T` without bounds, while after this change that no longer compiles without adding `T: Iterator` there as well.

Reply:
The function is `pub(in crate::iter)` so not externally public API, and all internal callers use it with `T: Iterator`.

Is this still a problem?

Reply:
Oh I'm sorry, looks like I misread your code. Yes, this is not a problem.

Reply:
No worries :)